### ISSUE TITLE
Abnormal long execution time when running "hello world"

### ISSUE THREAD

**Issue by hungryzzz** [wasmerio/wasmer#4201]
### Describe the bug

Hi, I build the latest `wasmer` and find a very long execution time (computed by `time` tool) even when running simple "hello world"(as followed).

- `Wasmer`: 2.466s
- `Wasmtime`: 0.015s

```c
#include <stdio.h>

int main() {
  printf("hello world\n");
  return 0;
}
```

I just insert some instrumentation rust code(as follow) in the `wasmer` project to collect which part of code take the most time, and the result is as follows. I find that `wasmer` spents too much time on `PluggableRuntime::new()`.
https://github.com/wasmerio/wasmer/blob/b03102d453547bb5d0fc0c605f57e6d4566f464a/lib/cli/src/commands/run/wasi.rs#L262

```shell
# output
⠴ Initializing the WebAssembly VM                                                                                                                                 
Time elapsed in PluggableRuntime::new is: 2.332459878s
hello world
Time elapsed in run_with_store_async() is: 1.510411ms
Time elapsed in execute_inner() is: 2.337409944s
```

```rust
// Instrumentation code
let start = Instant::now();  // Time begin

let mut rt = PluggableRuntime::new(Arc::new(TokioTaskManager::new(rt_or_handle.into())));

let duration = start.elapsed();  // Time end
println!("Time elapsed in prepare_runtime() is: {:?}", duration);  // print the duration
```

However, when I change another machine to repeat the above steps, the execution time is normal (almost the same as `wasmtime`), so I think maybe this bug is related to the physical machine. But I don't know how to do the further debug.

```sh
-> wasmer -vV
wasmer 4.1.2 (b03102d 2023-09-05)
binary: wasmer-cli
commit-hash: b03102d453547bb5d0fc0c605f57e6d4566f464a
commit-date: 2023-09-05
host: x86_64-unknown-linux-gnu
compiler: singlepass,cranelift

-> rustc -vV
rustc 1.72.0 (5680fa18f 2023-08-23)
binary: rustc
commit-hash: 5680fa18feaa87f3ff04063800aec256c3d4b4be
commit-date: 2023-08-23
host: x86_64-unknown-linux-gnu
release: 1.72.0
LLVM version: 16.0.5

-> uname -a
Linux xxx 5.15.0-69-generic #76-Ubuntu SMP Fri Mar 17 17:19:29 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```

### Steps to reproduce

```shell
wasmer hello.wasm
```

[hello.wasm.txt](https://github.com/wasmerio/wasmer/files/12521307/hello.wasm.txt)

**Comment by hungryzzz** [comment]
I've also tried another two machine to run the "hello world" case using `wasmer`(all the versions are the same as above), the execution result is as follow.

It seems that the execution time is related to some features of physical machine.

```shell
# machine A

-> uname -a
Linux xxx 5.15.0-69-generic #76-Ubuntu SMP Fri Mar 17 17:19:29 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

-> time ../wasmer/target/release/wasmer hello.wasm
hello world
../wasmer/target/release/wasmer hello.wasm  22.44s user 0.69s system 2156% cpu 1.072 total

-> time ../wasmtime/target/release/wasmtime hello.wasm
hello world
../wasmtime/target/release/wasmtime hello.wasm  0.00s user 0.00s system 100% cpu 0.009 total
```

```shell
# machine B

-> uname -a
Linux xxx 5.15.0-69-generic #76-Ubuntu SMP Fri Mar 17 17:19:29 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

-> time ../wasmer/target/release/wasmer hello.wasm
hello world
../wasmer/target/release/wasmer hello.wasm  143.59s user 3.07s system 4769% cpu 3.075 total

-> time ../wasmtime/target/release/wasmtime hello.wasm
hello world
../wasmtime/target/release/wasmtime hello.wasm  0.01s user 0.00s system 100% cpu 0.008 total
```

**Comment by ptitSeb** [comment]
that's indeed very long. Can you try again with latest 4.2.0 that just went out yesterday?

**Comment by hungryzzz** [comment]
@ptitSeb I've try again with latest commit(cf51f38568bea3de369d05514640872589dd6c5e) of `wasmer`, the result is almost the same as above.

```shell
# the first mentioned machine
-> wasmer hello.wasm
⠴ Initializing the WebAssembly VM                                                                                                                                 Time elapsed in PluggableRuntime::new is: 2.368649716s
hello world
Time elapsed in run_with_store_async() is: 1.672804ms
Time elapsed in execute_inner() is: 2.374535564s
```

```shell
# machine A
-> time ../wasmer/target/release/wasmer hello.wasm
hello world
../wasmer/target/release/wasmer hello.wasm  22.17s user 0.77s system 2082% cpu 1.101 total
```

```shell
# machine B
-> time ../wasmer/target/release/wasmer hello.wasm
hello world
../wasmer/target/release/wasmer hello.wasm  146.01s user 3.15s system 4762% cpu 3.132 total
```

**Comment by ptitSeb** [comment]
Can you run with `RUST_LOG=trace wasmer run hello.wasm` and put the log here.

**Comment by ptitSeb** [comment]
where does this line
```
Time elapsed in PluggableRuntime::new is: 2.485054208s
```
came from?

**Comment by Arshia001** [comment]
@hungryzzz are you running the official wasmer CLI? Also, it looks like `wasmer_cli::commands::run: enter` and `wasmer_cli::commands::run: close` are just 4.5 milliseconds apart, which should be the time wasmer took to execute the module. Did wasmer take 20 seconds to run? Was that delay before or after the logs started showing up?

**Comment by hungryzzz** [comment]
> where does this line
> 
> ```
> Time elapsed in PluggableRuntime::new is: 2.485054208s
> ```
> 
> came from?

@ptitSeb I add some instrumentation code in `wasmer` as I wrote in the beginning.

```rust
// Instrumentation code
let start = Instant::now();  // Time begin

let mut rt = PluggableRuntime::new(Arc::new(TokioTaskManager::new(rt_or_handle.into())));

let duration = start.elapsed();  // Time end
println!("Time elapsed in prepare_runtime() is: {:?}", duration);  // print the duration
```

**Comment by hungryzzz** [comment]
> @hungryzzz are you running the official wasmer CLI? Also, it looks like `wasmer_cli::commands::run: enter` and `wasmer_cli::commands::run: close` are just 4.5 milliseconds apart, which should be the time wasmer took to execute the module. Did wasmer take 20 seconds to run? Was that delay before or after the logs started showing up?

Hi, I add some instrumentation code in `wasmer` to record the execution time. The result is as followed.

```
# output
⠴ Initializing the WebAssembly VM                                                                                                                                 
Time elapsed in PluggableRuntime::new is: 2.332459878s
hello world
Time elapsed in run_with_store_async() is: 1.510411ms
Time elapsed in execute_inner() is: 2.337409944s
```

According to the logs, I find that `let mut rt = PluggableRuntime::new(Arc::new(TokioTaskManager::new(rt_or_handle.into())));` in `lib/cli/src/commands/run/wasi.rs` will spend most of time.

**Comment by Arshia001** [comment]
@hungryzzz thanks, that's very helpful.

**Comment by hungryzzz** [comment]
Actually, I found that the current version of `wasmer` is slower than `wasmer 4.0.0`. 
![wasmer 4 0 0](https://github.com/wasmerio/wasmer/assets/32137313/7df664c0-b4a2-4b14-98f1-a797453077f6)
![wasmer 4 2 2](https://github.com/wasmerio/wasmer/assets/32137313/1ea7e23f-a5b4-4bf7-85cf-4caefd5084f4)

**Comment by syrusakbary** [comment]
I just tested on latest `main` and it seems the issue is already resolved in Wasmer `4.3.0`.
Please feel free to reopen if it persists on your side
